### DISCUSSION TITLE
Trying to understand a possible CI integration

### DISCUSSION THREAD

**Discussion by whysthatso** [caronc/apprise-api#175] [Q&A]
I'm using woodpecker ci, a fork of drone ci, and i try to wrap my head around notifications.

I see two possibilities:

1. I have apprise-api running as a network-accessible service and during ci runs i POST to this api to send out notifictions, possibly via curl or any other similar tool or library

2. I somehow have apprise dockerized so i can run it as a woodpecker plugin and call it directly, with a configuration attached to it.

Can anybody advise or is anyone running a setup like that?

**Comment by caronc** [comment]
I personally use a dockerized container and park it behind an nginx proxy (with fail2ban).  this way i can secure everything with a login and password (using basicauth).

**Reply by mattpackwood** [reply]
Does that allow you remote access?

**Reply by whysthatso** [reply]
yes, that's what i do, too. i guess making it a ci-plugin is a worthwhile project, but nothing i can source at the moment.

if anybody is up for it, maybe have a look at this:  https://woodpecker-ci.org/docs/usage/plugins/creating-plugins
